config.go

```diff
@@ -156,7 +158,14 @@ func makeDockerV2S1Image(manifest docker.V2S1Manifest) (docker.V2Image, error) {
 				h.CreatedBy = fmt.Sprintf("%v", dcompat.ContainerConfig.Cmd)
 			}
 			h.EmptyLayer = dcompat.ThrowAway
+			id = dcompat.ID
 		}
+		// Skip this history item if it shares the ID of the last one
+		// that we saw, since the image library will do the same.
+		if id != "" && id == lastID {
```

Comment:
Is id capable of still being "" without being  an error?  If id == "" and last ID == "", should that be a continue?

Reply:
That's a good point.  The decode failing could reasonably be a reason to bail on importing the image, in which case `id` would always be set.  I'll rework that bit.